Visibility refactor in #868 broke compilation under MinGW GCC

Hi,

In #868 the visibility system was completely overhauled. Unfortunately, the assumption that `default`-defined destructors can be manually exported breaks building with MinGW GCC.

```
In file included from E:/krita-win/b_deps_msys/ext_openexr/ext_openexr-prefix/src/ext_openexr/OpenEXR/IlmImf/ImfKeyCodeAttribute.h:47,
                 from E:/krita-win/b_deps_msys/ext_openexr/ext_openexr-prefix/src/ext_openexr/OpenEXR/IlmImf/ImfStandardAttributes.h:63,
                 from E:\krita-win\b_deps_msys\ext_openexr\ext_openexr-prefix\src\ext_openexr\OpenEXR\exrenvmap\readInputImage.cpp:46:
E:/krita-win/b_deps_msys/ext_openexr/ext_openexr-prefix/src/ext_openexr/OpenEXR/IlmImf/ImfKeyCode.h:124:5: error: function 'Imf_2_5::KeyCode::~KeyCode()' definition is marked dllimport
  124 |     ~KeyCode() = default;
      |   
```

OpenEXR version: `2.5.5`
GCC version: `g++.exe (Rev9, Built by MSYS2 project) 11.2.0`

Probably cc @cary-ilm because it seems a validation of #358.

--- a/OpenEXR/IlmImf/ImfKeyCode.h
+++ b/OpenEXR/IlmImf/ImfKeyCode.h
@@ -120,7 +120,6 @@ class KeyCode
 
     IMF_EXPORT
     KeyCode (const KeyCode &other);
-    IMF_EXPORT
     ~KeyCode() = default;
     IMF_EXPORT
     KeyCode & operator = (const KeyCode &other);

